tests/internal/neighbor/test_neighbor.py

```diff
@@ -0,0 +1,75 @@
+from typing import cast
+import pytest
+import numpy as np
+from sklearn.neighbors import NearestNeighbors
+
+from cleanlab.internal.neighbor import features_to_knn
+from cleanlab.internal.neighbor.neighbor import knn_to_knn_graph
+
+
+@pytest.mark.parametrize(
+    "N",
+    [2, 10, 100, 101],
+    ids=lambda x: f"N={x}",
+)
+@pytest.mark.parametrize(
+    "M",
+    [2, 3, 4, 5, 10, 50, 100],
+    ids=lambda x: f"M={x}",
+)
+def test_features_to_knn(N, M):
+
+    features = np.random.rand(N, M)
```

Comment:
can we make the last 10 entries of features all identical, just to include an edge-case in this test already?